### ISSUE TITLE
This is ideas.

### ISSUE THREAD

**Issue by williamstorm** [dularion/streama#221]
Subscription Center - Like netflix allow people to purchase access
Themes - Allow themes to be made / easy settings in dashboard
Downloads - allow a certain user role (subscribers mainly) to download the files directly
Adverts - allow adverts before playing the videos

will have more soon.

**Comment by z3ntu** [comment]
1. Subscription Center: I don't think that you have any rights in distributing files for more than personal use and I don't think this should be more than a private application for personal use.
2. Themes: Good idea
3. Downloads: Good idea too.
4. Adverts: See (1).

**Comment by williamstorm** [comment]
but what if its self made tutorials or things that help people and charge subscriptions?

**Comment by terbolous** [comment]
- Separating library from application, this could allow for using your NAS or other small machine as a library server, and use something else for the actual webapp/frontend. I'm thinking the library should serve as an API and stream out the video as requested by the frontend

- Working forward from the previous idea, allow for separate transcoding hosts. When you reach a certain amount of users that needs transcoding (when that is in place) you might not have enough processing power on your file host (or library if that is implemented), and allowing the usage of a separate transcoding host would be great.

The frontend should have a list of available transcoding hosts (the admin could/should specify a max # of concurrent transcodings), and schedule one as appropriate, whereas the transcoder retrieves the content from a library server and streams the transcoded result to the frontend

- The frontend should support several library backends and transcoding hosts, but also support doing it all itself

**Comment by ManuGithubSteam** [comment]
> Separating library from application, this could allow for using your NAS or other small machine as a library server, and use something else for the actual webapp/frontend. I'm thinking the library should serve as an API and stream out the video as requested by the frontend

This would mean i need a client program who runs on my pc and connects to the api on my nas. Right ? If that is so, i'm not in favor of this. I choose streama because i can avoid client programs (like kodi)
Could you elaborate more on your idea? 

Transcoding takes time but when your whole library is in the right format (which will take me like 3-6 months) then adding a few new films is not the problem anymore. So while i think the idea with many transcoding hosts is good for the initial conversation of the movies, i doubt it has any real use after your bulk of files is encoded in the right format. 
After all a decent machine can convert 5 movies in one day, so at least for me i don't see the application after the initial transcoding.

**Comment by terbolous** [comment]
@ManuGithubSteam No, that is not what I meant. I meant that there would be several web applications, one for the library for which the frontend talks to, and one for the frontend that displays the webpage users would watch. There shouldn't be anything wrong with running them both on the same server/box, or separating them depending on your needs/wishes.

Did that clarify? :-)

Having to store stuff in various formats is both time and disk space consuming, and unnecessary in my opinion. I'm not even sure if there is 1 format that is universally working across devices?

**Comment by phi1-h** [comment]
@williamstorm I like the idea of an download option!
@terbolous I agree on separating library and frontend. I think the option to add multiple library backends is great and enables private groups to gather their movies in one place! I do agree on only enabling as less formats as neccesary, as well.

**Comment by dularion** [comment]
See this issue for subscriptions #602
